transactionparser/src/orderbookchanges.js

```diff
@@ -0,0 +1,82 @@
+var _ = require('lodash');
+var utils = require('./utils');
+var BigNumber = require('bignumber.js');
+
+/**
+ * Computes the complete list of every Offer that changed in the ledger
+ * as a result of the given transaction.
+ *
+ *  @param {Object} metadata
+ *
+ */
+function parseOrderbookChanges(metadata) {
+  var orderbookChanges = utils.normalizeNodes(metadata).map(function(node) {
+    if (node.entryType === 'Offer') {
+      return parseOrderbookChange(node);
+    }
+  });
+
+  return _.compact(orderbookChanges);
+}
+
+function parseOrderbookChange(node) {
+  var diffType = node.diffType;
+  var status = parseOrderStatus(node);
+
+  return {
+    account: node.finalFields.Account,
+    taker_pays: parseChangeAmount(node, 'TakerPays'),
+    taker_gets: parseChangeAmount(node, 'TakerGets'),
+    sequence: node.finalFields.Sequence,
+    status: status
+  };
+
+  function parseChangeAmount(node,type) {
+    var previousAmount;
+    var finalAmount;
+    var changeAmount;
+    var finalValue;
+    var prevValue;
```

Comment:
I don't know if we should follow that advice. I'm guessing they wrote that under the assumption that you are not using a linter. Are there any potential issues with hoisting that a linter won't catch? A linter should warn you if you ever use a variable before it is declared, so I can't think of any. Keeping it in the tightest possible scope is considered best practice in other languages, increases cohesion, and reduces lines of code. And hopefully future versions of javascript will implement block-level scoping.